File: UI/src/common/Enumeration.ts
/**
 * Enum mã lỗi chương trình
 */
export enum EnumApplicationErrorCode {
    /// Lỗi trùng thông tin đăng ký (SĐT, mã nhân viên)
    DuplicateLoginInfo = 4001
}

/**
 * Edit mode
 */
export enum EnumEditMode {
    Add = 1,

    Edit = 2,

    Delete = 3
}

/**
 * Enum kiểu dữ liệu
 */
export enum EnumDataType {
    String = 1,

    Integer = 2,

    Boolean = 3
}

export enum EnumUserType {
    Employee = 0,

    Customer = 1
}

/**
 * Enum người gửi tin nhắn trong chat bot
 */
export enum EnumChatbotSender {
    /// Bot
    Bot = 0,

    /// Người dùng
    User = 1
}

/**
 * Enum trạng thái đặt bàn
 */
export enum EnumReservationStatus {
    /// Đang chờ xác nhận
    Pending = 1,

    /// Đã xác nhận
    Confirmed = 2,

    /// Đã huỷ
    Canceled = 3,

    /// Đã nhận bàn
    Received = 4
}

/**
 * Enum trạng thái làm việc của nhân viên
 */
export enum EnumEmployeeWorkStatus {
    // Đang làm việc
    Active = 0,

    // Thử việc
    Probation = 1,

    // Nghỉ phép
    OnLeave = 2,

    // Nghỉ việc
    Terminated = 3
}

/**
 * Enum loại món
 */
export enum EnumMenuItemCategory {
    /// Khai vị
    Appetizers = 0,

    /// Món chính
    MainCourse = 1,

    /// Tráng miệng
    Dessert = 2,

    /// Đồ uống
    Drink = 3
}

/**
 * Enum trạng thái order
 */
export enum EnumOrderStatus {
    Active = 1,

    Paid = 2,

    Canceled = 3
}

/**
 * Enum ngày trong tuần
 */
export enum EnumDayOfWeek {
    Sunday = 0,

    Monday = 1,

    Tuesday = 2,

    Wednesday = 3,

    Thursday = 4,

    Friday = 5,

    Saturday = 6,

    Holiday = 7
}

export enum EnumPaymentMethod {
    Cash = 0,
    
    Transaction = 1,

    Card = 2
}

/**
 * Trạng thái bàn
 */
export enum EnumTableStatus {
    // Còn trống
    Available = 0,

    // Hết chỗ
    Occupied = 1,

    // Đã được đặt chỗ
    Reserved = 2
}

/**
 * Enum trạng thái yêu cầu tạo món custom
 */
export enum EnumCustomMenuRequestStatus {
    Pending = 0,

    Approved = 1,

    Rejected = 2
}

class EnumerationInterface {
    EnumUserType: typeof EnumUserType = EnumUserType;
    EnumChatbotSender: typeof EnumChatbotSender = EnumChatbotSender;
    EnumMenuItemCategory: typeof EnumMenuItemCategory = EnumMenuItemCategory;
    EnumEditMode: typeof EnumEditMode = EnumEditMode;
    EnumReservationStatus: typeof EnumReservationStatus = EnumReservationStatus;
    EnumRole: typeof EnumEmployeeWorkStatus = EnumEmployeeWorkStatus;
    EnumOrderStatus: typeof EnumOrderStatus = EnumOrderStatus;
    EnumDayOfWeek: typeof EnumDayOfWeek = EnumDayOfWeek;
    EnumEmployeeWorkStatus: typeof EnumEmployeeWorkStatus = EnumEmployeeWorkStatus;
    EnumPaymentMethod: typeof EnumPaymentMethod = EnumPaymentMethod;
    EnumTableStatus: typeof EnumTableStatus = EnumTableStatus;
    EnumCustomMenuRequestStatus: typeof EnumCustomMenuRequestStatus = EnumCustomMenuRequestStatus;
}

export default EnumerationInterface;

File: API/ML.Common/Enumeration.cs
namespace API.ML.Common
{
    /// <summary>
    /// Enum mã lỗi chương trình
    /// </summary>
    public enum EnumApplicationErrorCode
    {
        /// <summary>
        /// Lỗi trùng thông tin đăng nhập (SĐT, mã NV,...)
        /// </summary>
        DuplicateLoginInfo = 4001,

        /// <summary>
        /// Sai thông tin đăng nhập
        /// </summary>
        InvalidLoginInfo = 4002
    }

    public enum EnumTimeFilter
    {
        ByDay = 0,

        ByWeek = 1,

        ByMonth = 2,

        ByYear = 3,

        Custom = 4
    }

    /// <summary>
    /// Edit mode
    /// </summary>
    public enum EnumEditMode
    {
        Add = 1,

        Edit = 2,

        Delete = 3
    }

    /// <summary>
    /// Enum kiểu dữ liệu
    /// </summary>
    public enum EnumDataType
    {
        String = 1,

        Integer = 2,

        Boolean = 3
    }

    /// <summary>
    /// Enum loại người dùng
    /// </summary>
    public enum EnumUserType
    {
        Employee = 0,

        Customer = 1
    }

    /// <summary>
    /// Enum quyền nhân viên
    /// </summary>
    public enum EnumEmployeeWorkStatus
    {
        /// <summary>
        /// Đang làm việc
        /// </summary>
        Active = 0,

        /// <summary>
        /// Thử việc
        /// </summary>
        Probation = 1,

        /// <summary>
        /// Đang nghỉ phép
        /// </summary>
        OnLeave = 2,

        /// <summary>
        /// Nghỉ việc
        /// </summary>
        Terminated = 3
    }

    public enum EnumOrderStatus
    {
        Active = 1,
        
        Paid = 2,

        Canceled = 3
    }

    /// <summary>
    /// Enum người gửi tin nhắn chatbot
    /// </summary>
    public enum EnumChatbotSender
    {
        /// <summary>
        /// Bot
        /// </summary>
        Bot = 0,

        /// <summary>
        /// Người dùng
        /// </summary>
        User = 1
    }

    /// <summary>
    /// Enum trạng thái đặt bàn
    /// </summary>
    public enum EnumReservationStatus
    {
        /// <summary>
        /// Chờ xác nhận
        /// </summary>
        Pending = 1,

        /// <summary>
        /// Đã xác nhận
        /// </summary>
        Confirmed = 2,

        /// <summary>
        /// Đã huỷ
        /// </summary>
        Cancelled = 3,

        /// <summary>
        /// Đã nhận bàn
        /// </summary>
        Received = 4
    }

    /// <summary>
    /// Enum cách hiển thị màn thực đơn cho khách hàng
    /// </summary>
    public enum EnumDisplayMenuScreenForCustomerType
    {
        /// <summary>
        /// Theo ảnh
        /// </summary>
        ByImages = 0,

        /// <summary>
        /// Theo món thiết lập
        /// </summary>
        ByItems = 1
    }

    /// <summary>
    /// Enum cách hiển thị màn thực đơn theo món thiết lập cho khách hàng
    /// </summary>
    public enum EnumDisplayMenuScreenByItemsForCustomerType
    {
        /// <summary>
        /// Toàn bộ món
        /// </summary>
        All = 0,

        /// <summary>
        /// Tuỳ chọn
        /// </summary>
        Custom = 1
    }

    /// <summary>
    /// Enum hình thức thanh toán
    /// </summary>
    public enum EnumPaymentMethod
    {
        /// <summary>
        /// Tiền mặt
        /// </summary>
        Cash = 0,

        /// <summary>
        /// Chuyển khoản
        /// </summary>
        Transaction = 1,

        /// <summary>
        /// Thẻ
        /// </summary>
        Card = 2
    }

    public enum EnumTableStatus
    {
        /// <summary>
        /// Còn trống
        /// </summary>
        Available = 0,

        /// <summary>
        /// Hết chỗ
        /// </summary>
        Occupied = 1,

        /// <summary>
        /// Đã được đặt chỗ
        /// </summary>
        Reserved = 2
    }

    /// <summary>
    /// Enum trạng thái yêu cầu tạo món custom
    /// </summary>
    public enum EnumCustomMenuRequestStatus
    {
        Pending = 0,

        Approved = 1,

        Rejected = 2
    }

    /// <summary>
    /// Enum loại chương trình khuyến mại
    /// </summary>
    public enum EnumPromotionType
    {
        /// <summary>
        /// Giảm giá món
        /// </summary>
        ItemDiscount = 0,

        /// <summary>
        /// Giảm giá order
        /// </summary>
        OrderDiscount = 1
    }

    /// <summary>
    /// Enum cách tính giá trị khuyến mại
    /// </summary>
    public enum EnumPromotionValueType
    {
        /// <summary>
        /// Theo phần trăm
        /// </summary>
        ByPercent = 0,

        /// <summary>
        /// Theo số tiền
        /// </summary>
        ByAmount = 1
    }
}
